### PR TITLE
Clarify inproc propagation of TraceId & TraceState

### PR DESCRIPTION
**Pull request by Oberon00**
In my understanding, these are minimum requirements to be able to implement the W3C TraceContext spec correctly.

### CODE REVIEW

File: specification/api-tracing.md [hunk 1]
```diff
@@ -255,6 +255,8 @@ trace. A span is said to be a _root span_ if it does not have a parent. Each
 trace includes a single root span, which is the shared ancestor of all other
 spans in the trace. Implementations MUST provide an option to create a `Span` as
 a root span, and MUST generate a new `TraceId` for each root span created.
+For a Span with a parent, the `TraceId` MUST be the same as the parent.
+Also, the child span MUST inherit all `TraceState` values of its parent by default.
```
Comment (yurishkuro) [inline specification/api-tracing.md]: This feels too specific.  W3C `tracestate` is just one of many possible propagation formats, the API does not depend on it. Can this be phrased in a more general form?

Comment (Oberon00) [reply specification/api-tracing.md]: The TraceState I mean here is not a format, it is a field of the SpanContext (See https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/api-tracing.md#spancontext).

Comment (SergeyKanzhelev) [reply specification/api-tracing.md]: This concern seems to be resolved.

**Comment by SergeyKanzhelev** [comment]
this is a small and important clarification of an existing spec. Non controversial. Merging as it was active 2+ days and have enough signoffs

**Pull request merged**